src/web_interface/components/database_routes.py

```diff
@@ -152,18 +151,17 @@ def _build_search_query(self):
             self._add_hash_query_to_query(query, request.form['hash_value'])
         if 'tags' in request.form and request.form['tags']:
             self._add_tag_query(query)
-        print(query)
         return json.dumps(query)
 
     @staticmethod
     def _add_multiple_choice(query, key):
-        new_query = [{key: name} for name in dict(request.form.lists())[key]]
-        query.setdefault('$or', []).extend(new_query)
+        for name in dict(request.form.lists())[key]:
+            query[key] = {"$in": [name]}
```

Comment:
This would replace the key in each iteration of the for loop.
Instead, it would have to be something like this:
```suggestion
        query[key] = {"$in": [name for name in dict(request.form.lists())[key]]}
```